typescript/phoenix-sdk/src/utils/seatManager.ts

```diff
@@ -0,0 +1,187 @@
+
+import {
+  createEvictSeatInstruction,
+  createClaimSeatInstruction,
+  PROGRAM_ID as PSM_PROGRAM_ID,
+} from "../../psm-src/index";
+
+/**
+ * Deserializes seat manager data from a given buffer and returns a SeatManagerData object
+ *
+ * @param data The data buffer to deserialize
+ */
+export function deserializeSeatManagerData(data: Buffer): SeatManagerData {
+  const [seatManagerData, _] = seatManagerBeet.deserialize(data, 0);
+  return seatManagerData;
+}
+
+export type SeatManagerData = {
+  market: PublicKey;
+  authority: PublicKey;
+  successor: PublicKey;
+  numMakers: beet.bignum;
+  _headerPadding: beet.bignum[] /* size: 11 */;
+  designatedMarketMaker: PublicKey[];
+  _dmmPadding: beet.bignum[] /* size: 128 */;
+};
+
+export const seatManagerBeet = new beet.BeetArgsStruct<SeatManagerData>(
+  [
+    ["market", beetSolana.publicKey],
+    ["authority", beetSolana.publicKey],
+    ["successor", beetSolana.publicKey],
+    ["numMakers", beet.u64],
+    ["_headerPadding", beet.uniformFixedSizeArray(beet.u64, 11)],
+    [
+      "designatedMarketMaker",
+      beet.uniformFixedSizeArray(beetSolana.publicKey, 128),
+    ],
+    ["_dmmPadding", beet.uniformFixedSizeArray(beet.u128, 128)],
+  ],
+  "SeatManagerData"
+);
+
+/**
+ * Find the seat manager's address
+ * @param marketPubkey The market's address
+ */
+export function getSeatManagerAddress(market: PublicKey): PublicKey {
+  return PublicKey.findProgramAddressSync(
+    [market.toBuffer()],
+    PSM_PROGRAM_ID
+  )[0];
+}
+
+/**
+ * Get the seat deposit collector's address
+ * @param marketPubkey The market's address
+ */
+export function getSeatDepositCollectorAddress(market: PublicKey): PublicKey {
+  return PublicKey.findProgramAddressSync(
+    [market.toBuffer(), Buffer.from("deposit")],
+    PSM_PROGRAM_ID
+  )[0];
+}
+
+/**
+ * Returns an instruction to claim a seat on a market, via the Phoenix Seat Manager
+ *
+ * @param market The market's address
+ * @param trader The trader's address
+ */
+export function getClaimSeatIx(
+  market: PublicKey,
+  trader: PublicKey
+): TransactionInstruction {
+  const seatManager = getSeatManagerAddress(market);
+  const seatDepositCollector = getSeatDepositCollectorAddress(market);
+  const seat = getSeatAddress(market, trader);
+  const logAuthority = getLogAuthorityAddress();
+
+  const claimSeatAccounts = {
+    phoenixProgram: PROGRAM_ID,
+    logAuthority,
+    market,
+    seatManager,
+    seatDepositCollector,
+    trader,
+    payer: trader,
+    seat,
+  };
+
+  return createClaimSeatInstruction(claimSeatAccounts);
+}
+
+/**
+ * Returns an instruction to evict a seat on a market, via the Phoenix Seat Manager
+ * Evict seat is only allowed if the trader state is full for a given market, unless performed by the seat manager authority
+ *
+ * @param market The market's address
+ * @param marketData The market's data, containing base and quote mint information
+ * @param trader The trader's address
```

Comment:
docs are incomplete here. it's also unclear from the doc who is getting evicted--i assume it's `trader`? how did we find the trader to evict? we should compute that too

Reply:
this method should first find a seat to evict and not take `trader` as an input

Reply:
I think we should have this as the instruction builder and then a function which performs the "find a trader to evict" and then creates an instruction, with this instruction builder. That way, we have the most flexibility (such as allowing eviction of a specific trader) with existing helper functions. Let me know if you disagree.

Reply:
lg